zcash_primitives/src/consensus.rs

```diff
@@ -484,6 +497,70 @@ impl BranchId {
         // Sprout rules apply before any network upgrade
         BranchId::Sprout
     }
+
+    pub fn min_height<P: Parameters>(&self, params: &P) -> Option<BlockHeight> {
```

Comment:
Document this.

Reply:
Actually, replace these APIs with an `epoch_range()` API that returns `Option<(Bound<BlockHeight>, Bound<BlockHeight>)>`:

- `None` indicates that the epoch will never occur because the corresponding network upgrade has no activation height.
- The bounds tuple implements `RangeBounds<BlockHeight>` and can therefore be used as a single range object.